frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/FieldRow/FieldRowContainer.tsx

```diff
@@ -191,12 +197,31 @@ export const FieldRowContainer = ({
   }, [handleBuilderClick, isMobile])
 
   const handleDuplicateClick = useCallback(() => {
-    // Duplicate button should be hidden if field
-    // is not yet created, but guard here just in case
-    if (stateData.state !== FieldBuilderState.CreatingField) {
-      duplicateFieldMutation.mutate(field._id)
+    // Duplicate button should be hidden if field is not yet created, but guard here just in case
+    if (stateData.state === FieldBuilderState.CreatingField) return
+    // Disallow duplicating attachment fields if after the dupe, the filesize exceeds the limit
+    if (field.fieldType === BasicField.Attachment) {
+      if (!form) return
```

Comment:
Should we put this at the top of the function? Ideally no duplication will happen if there is no form anyways

Reply:
yesss moved in [4bdcda2](https://github.com/opengovsg/FormSG/pull/4699/commits/4bdcda2b8c579eca2c630ce339e255bae3a9991d)